src/http_connection.cpp

```diff
@@ -148,7 +148,7 @@ void http_connection::get(std::string const& url, time_duration timeout, int pri
 
 	if (ec)
 	{
-		m_timer.get_io_service().post(std::bind(&http_connection::callback
```

Comment:
iirc, all calls to `post()` like this could be changed to use the free function version of `post()`. like this: `post(m_timer, std::bind(...))`

Reply:
I'll need your help on this one. From my understanding, you want to call:
```
template <typename ExecutionContext, typename CompletionToken>
inline BOOST_ASIO_INITFN_RESULT_TYPE(CompletionToken, void()) post(
    ExecutionContext& ctx, BOOST_ASIO_MOVE_ARG(CompletionToken) token,
    typename enable_if<is_convertible<
      ExecutionContext&, execution_context&>::value>::type*)
```
However, this fails (probably because of the enable_if argument) on:
```

http_connection.cpp: In member function ‘void libtorrent::http_connection::get(const string&, libtorrent::time_duration, int, const libtorrent::aux::proxy_settings*, int, const string&, const boost::optional<boost::asio::ip::address>&, libtorrent::resolver_flags, const string&, libtorrent::i2p_connection*)’:
http_connection.cpp:152:27: error: no matching function for call to ‘post(libtorrent::deadline_timer&, std::_Bind_helper<false, void (libtorrent::http_connection::*)(boost::system::error_code, libtorrent::span<char>), std::shared_ptr<libtorrent::http_connection>&, boost::system::error_code&, libtorrent::span<char> >::type)’
    , me, ec, span<char>{}));
                           ^
In file included from /home/fabrice/buildroot-test/buildroot/output/host/sh4-buildroot-linux-uclibc/sysroot/usr/include/boost/asio/post.hpp:113:0,
                 from /home/fabrice/buildroot-test/buildroot/output/host/sh4-buildroot-linux-uclibc/sysroot/usr/include/boost/asio/basic_socket.hpp:28,
                 from /home/fabrice/buildroot-test/buildroot/output/host/sh4-buildroot-linux-uclibc/sysroot/usr/include/boost/asio/basic_socket_acceptor.hpp:19,
                 from /home/fabrice/buildroot-test/buildroot/output/host/sh4-buildroot-linux-uclibc/sysroot/usr/include/boost/asio/ip/tcp.hpp:19,
                 from ../include/libtorrent/socket.hpp:53,
                 from ../include/libtorrent/http_connection.hpp:51,
                 from http_connection.cpp:33:
/home/fabrice/buildroot-test/buildroot/output/host/sh4-buildroot-linux-uclibc/sysroot/usr/include/boost/asio/impl/post.hpp:79:63: note: candidate: template<class ExecutionContext, class CompletionToken> typename boost::asio::async_result<typename std::decay<_Arg>::type, void()>::return_type boost::asio::post(ExecutionContext&, CompletionToken&&, typename std::enable_if<std::is_convertible<ExecutionContext&, boost::asio::execution_context&>::value>::type*)
 inline BOOST_ASIO_INITFN_RESULT_TYPE(CompletionToken, void()) post(
                                                               ^~~~
/home/fabrice/buildroot-test/buildroot/output/host/sh4-buildroot-linux-uclibc/sysroot/usr/include/boost/asio/impl/post.hpp:79:63: note:   template argument deduction/substitution failed:
/home/fabrice/buildroot-test/buildroot/output/host/sh4-buildroot-linux-uclibc/sysroot/usr/include/boost/asio/impl/post.hpp: In substitution of ‘template<class ExecutionContext, class CompletionToken> typename boost::asio::async_result<typename std::decay<_Arg>::type, void()>::return_type boost::asio::post(ExecutionContext&, CompletionToken&&, typename std::enable_if<std::is_convertible<ExecutionContext&, boost::asio::execution_context&>::value>::type*) [with ExecutionContext = boost::asio::basic_waitable_timer<std::chrono::_V2::system_clock>; CompletionToken = std::_Bind<void (libtorrent::http_connection::*(std::shared_ptr<libtorrent::http_connection>, boost::system::error_code, libtorrent::span<char>))(boost::system::error_code, libtorrent::span<char>)>]’:
http_connection.cpp:152:27:   required from here
/home/fabrice/buildroot-test/buildroot/output/host/sh4-buildroot-linux-uclibc/sysroot/usr/include/boost/asio/impl/post.hpp:79:63: error: no type named ‘type’ in ‘struct std::enable_if<false, void>’
/home/fabrice/buildroot-test/buildroot/output/host/sh4-buildroot-linux-uclibc/sysroot/usr/include/boost/asio/impl/post.hpp:70:56: note: candidate: template<class Executor, class CompletionToken> typename boost::asio::async_result<typename std::decay<_Arg>::type, void()>::return_type boost::asio::post(const Executor&, CompletionToken&&, typename std::enable_if<boost::asio::is_executor<Executor>::value>::type*)
 BOOST_ASIO_INITFN_RESULT_TYPE(CompletionToken, void()) post(
                                                        ^~~~
/home/fabrice/buildroot-test/buildroot/output/host/sh4-buildroot-linux-uclibc/sysroot/usr/include/boost/asio/impl/post.hpp:70:56: note:   template argument deduction/substitution failed:
/home/fabrice/buildroot-test/buildroot/output/host/sh4-buildroot-linux-uclibc/sysroot/usr/include/boost/asio/impl/post.hpp: In substitution of ‘template<class Executor, class CompletionToken> typename boost::asio::async_result<typename std::decay<_Arg>::type, void()>::return_type boost::asio::post(const Executor&, CompletionToken&&, typename std::enable_if<boost::asio::is_executor<Executor>::value>::type*) [with Executor = boost::asio::basic_waitable_timer<std::chrono::_V2::system_clock>; CompletionToken = std::_Bind<void (libtorrent::http_connection::*(std::shared_ptr<libtorrent::http_connection>, boost::system::error_code, libtorrent::span<char>))(boost::system::error_code, libtorrent::span<char>)>]’:
http_connection.cpp:152:27:   required from here
/home/fabrice/buildroot-test/buildroot/output/host/sh4-buildroot-linux-uclibc/sysroot/usr/include/boost/asio/impl/post.hpp:70:56: error: no type named ‘type’ in ‘struct std::enable_if<false, void>’
/home/fabrice/buildroot-test/buildroot/output/host/sh4-buildroot-linux-uclibc/sysroot/usr/include/boost/asio/impl/post.hpp:62:56: note: candidate: template<class CompletionToken> typename boost::asio::async_result<typename std::decay<_Tp>::type, void()>::return_type boost::asio::post(CompletionToken&&)
 BOOST_ASIO_INITFN_RESULT_TYPE(CompletionToken, void()) post(
                                                        ^~~~
/home/fabrice/buildroot-test/buildroot/output/host/sh4-buildroot-linux-uclibc/sysroot/usr/include/boost/asio/impl/post.hpp:62:56: note:   template argument deduction/substitution failed:
http_connection.cpp:152:27: note:   candidate expects 1 argument, 2 provided
    , me, ec, span<char>{}));
   
```